packages/server/dataloader/atlassianLoaders.ts

```diff
@@ -34,37 +33,53 @@ export const freshAtlassianAuth = (parent: RethinkDataLoader) => {
     async (keys) => {
       const results = await Promise.allSettled(
         keys.map(async ({userId, teamId}) => {
-          const atlassianAuth = await getAtlassianAuthByUserIdTeamId(userId, teamId)
-
-          if (!atlassianAuth?.refreshToken) {
+          const userAtlassianAuths = await getAtlassianAuthsByUserId(userId)
+          const atlassianAuthToRefresh = userAtlassianAuths.find(
+            (atlassianAuth) => atlassianAuth.teamId === teamId
+          )
+          if (!atlassianAuthToRefresh?.refreshToken) {
             // Not always an error! For suggested integrations, this won't exist.
             // sendToSentry(new Error('No atlassian access token exists for team member'), {
             //   userId,
             //   tags: {teamId}
             // })
             return null
           }
-          const {accessToken: existingAccessToken, refreshToken} = atlassianAuth
+          const {accessToken: existingAccessToken, refreshToken} = atlassianAuthToRefresh
           const decodedToken = existingAccessToken && (decode(existingAccessToken) as any)
           const now = new Date()
           const inAMinute = Math.floor((now.getTime() + 60000) / 1000)
           if (!decodedToken || decodedToken.exp < inAMinute) {
             const manager = await AtlassianServerManager.refresh(refreshToken)
             const {accessToken, refreshToken: newRefreshToken} = manager
-            atlassianAuth.accessToken = accessToken
-            atlassianAuth.updatedAt = now
-
-            if (newRefreshToken) {
-              atlassianAuth.refreshToken = newRefreshToken
-            }
+            const updatedRefreshToken = newRefreshToken ?? atlassianAuthToRefresh.refreshToken
+            // if user integrated the same Jira account with using different teams we need to update them as well
+            // reference: https://github.com/ParabolInc/parabol/issues/5601
+            const updatedSameJiraAccountAtlassianAuths = userAtlassianAuths
+              .filter(
+                (auth) =>
+                  auth.teamId !== teamId && auth.accountId === atlassianAuthToRefresh.accountId
+              )
+              .map((auth) => ({
+                ...auth,
+                accessToken,
+                refreshToken: updatedRefreshToken
+              }))
 
-            await upsertAtlassianAuth(atlassianAuth)
+            await upsertAtlassianAuths([
+              {
+                ...atlassianAuthToRefresh,
+                accessToken,
+                refreshToken: updatedRefreshToken
+              },
+              ...updatedSameJiraAccountAtlassianAuths
+            ])
```

Comment:
Added upsert to handle multiple items. Not sure if we want to remove `upsertAtlassianAuth` that takes only one item and simply use this one everywhere or leave both. Either way, I'm happy to improve if needed!

Reply:
i think this would be great since both queries are so similar. can you refactor to remove the one that returns only a single query? it would be a good general pattern imo to write queries to support multiple values instead of only single.

Reply:
Done!